backend/src/main/scala/bloop/Compiler.scala

```diff
@@ -606,26 +672,158 @@ object Compiler {
 
           cause match {
             case f: xsbti.CompileFailed =>
-              // We cannot guarantee reporter.problems == f.problems, so we aggregate them together
-              val reportedProblems = reporter.allProblemsPerPhase.toList
-              val rawProblemsFromReporter = reportedProblems.iterator.map(_.problem).toSet
-              val newProblems = f.problems().flatMap { p =>
-                if (rawProblemsFromReporter.contains(p)) Nil
-                else List(ProblemPerPhase(p, None))
-              }
-              val failedProblems = reportedProblems ++ newProblems.toList
+              val failedProblems = findFailedProblems(Some(f))
               val backgroundTasks =
                 toBackgroundTasks(backgroundTasksForFailedCompilation.toList)
-              Result.Failed(failedProblems, None, elapsed, backgroundTasks)
+              Result.Failed(failedProblems, None, elapsed, backgroundTasks, None)
             case t: Throwable =>
+              compileOut.bestEffortDirs.foreach {
+                case BestEffortDirs(buildDir, _) =>
+                  logger.info(
+                    "Unsuccessful best effort compilation. No best effort artifacts were created."
+                  )
+                  BloopPaths.delete(buildDir)
+              }
               t.printStackTrace()
               val backgroundTasks =
                 toBackgroundTasks(backgroundTasksForFailedCompilation.toList)
-              Result.Failed(Nil, Some(t), elapsed, backgroundTasks)
+              Result.Failed(Nil, Some(t), elapsed, backgroundTasks, None)
           }
       }
   }
 
+  /**
+   * Handles successful Best Effort compilation.
+   * Does not persist incrementalCompilation analysis.
+   */
+  def handleBestEffortSuccess(
```

Comment:
So this is needed so that we don't persist analysis? What happens if we do? Could explain a bit more in the comment?

Reply:
Oh sorry, this was a forget-me-not I ironically forgot about. I'm still going to have to run through this method, clean it up and make things less redundant - but basically now everything should at the very least work. About the comment, for now I reused a some methods from the way successful compilation is handled, however I had to remove handling of analysis.bin files from those for two reasons:
* to work it needs xsbti.compile.MiniSetup obtained from "successful" zinc compilation which we do not always have
* I guess we could reimplement it but I saw no purpose to it for those "best-effort" cases - it does not seem to affect anything there